### PR TITLE
Feat: Interactive Wrapper

### PR DESCRIPTION
**Pull request by jimhunty**
This PR gives access to all interactives on web via the previously built web components by the digital developers.

On Native, the interactives do no exist. This wrapper exposes the interactives via a Lambda function which controls which Interactives to show. This Lambda function is currently controlled by the digital developers.

A change is required on TPA to add id as a returned value to get this working end to end.

### CODE REVIEW

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/83c7737eeacef7dbddacae33e7ffa011a6037714/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/9cb44302435b310396a664f99ab3c481a62f01c3/index.html

File: packages/article/src/article-body/article-body-row.js [hunk 1]
```diff
@@ -28,6 +23,16 @@ const styles = StyleSheet.create({
...
+  },
+  interactiveContainer: {
+    paddingRight: spacing(2),
+    paddingLeft: spacing(2),
+    marginBottom: spacing(4)
+  },
+  primaryContainer: {
+    width: "100%",
+    flexDirection: "column",
+    paddingBottom: spacing(5)
```
Comment (colincclark) [inline packages/article/src/article-body/article-body-row.js]: could this lot be moved to the `styles/article-body` directory?

File: packages/article/src/styles/article-body/responsive.web.js [hunk 2]
```diff
@@ -41,6 +41,24 @@ export const Paragraph = styled.p`
   }
 `;
 
+export const InteractiveContainer = styled.div`
+  padding-left: ${spacing(2)};
+  padding-right: ${spacing(2)};
+  margin-bottom: ${spacing(4)};
+
+  @media (min-width: ${breakpoints.medium}px) {
+    width: 80.8%;
```
Comment (colincclark) [inline packages/article/src/styles/article-body/responsive.web.js]: still think these widths belong in the `styleguide` package. Have created https://github.com/newsuk/times-components/issues/1291

File: packages/interactive-wrapper/README.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+Or the tests for all platforms can be run
+
+```
+yarn test:all
+```
+
+Visit the official
+[storybook](http://components.thetimes.co.uk/?knob-Size%20of%20ad%20placeholder%3A=default&knob-Interactive=chapterHeading&knob-Interactive%20Wrapper=chapterHeading&selectedKind=Primitives%2FInteractive%20Wrapper&selectedStory=Interactive%20Wrapper&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybooks%2Fstorybook-addon-knobs)
+to see our available image templates.
```
Comment (colincclark) [inline packages/interactive-wrapper/README.md]: "interactive templates"

File: packages/interactive-wrapper/__tests__/interactive-wrapper.native.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+  it("handleNavigationStateChange should not open a link when the origin is the same", () => {
+    const component = setUpNavigationTest(() => Promise.resolve(true));
+    jest.spyOn(InteractiveWrapper, "openURLInBrowser");
+    component.handleNavigationStateChange({
+      url:
+        "http://cwfiyvo20d.execute-api.eu-west-1.amazonaws.com/same-origin-different-url"
+    });
+    expect(InteractiveWrapper.openURLInBrowser).not.toHaveBeenCalled();
+  });
+};
```
Comment (colincclark) [inline packages/interactive-wrapper/__tests__/interactive-wrapper.native.js]: 👍 nice

File: packages/interactive-wrapper/src/interactive-wrapper.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
+import React, { Component } from "react";
+import { Linking, Platform, View, WebView } from "react-native";
+import PropTypes from "prop-types";
+
+class InteractiveWrapper extends Component {
+  static postMessageBugWorkaround() {
+    return Platform.select({
```
Comment (colincclark) [inline packages/interactive-wrapper/src/interactive-wrapper.js]: would it be better to target platforms the RNW way with file extensions?

File: packages/interactive-wrapper/src/interactive-wrapper.js [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+  onLoadEnd() {
+    this.webview.postMessage("thetimes.co.uk", "*");
+  }
+
+  handleNavigationStateChange(data) {
+    if (
+      !data.url.includes("data:text/html") &&
+      data.url.includes("http") &&
+      !data.url.includes("cwfiyvo20d.execute-api.eu-west-1.amazonaws.com")
```
Comment (colincclark) [inline packages/interactive-wrapper/src/interactive-wrapper.js]: can this go in a constant somewhere?

File: packages/interactive-wrapper/README.md (hunk 3 above)
Comment (Belema) [reply packages/interactive-wrapper/README.md]: copy/paste

Comment (jimhunty) [reply packages/interactive-wrapper/README.md]: 🤦‍♂️

File: packages/article/src/styles/article-body/responsive.web.js (hunk 2 above)
Comment (jimhunty) [reply packages/article/src/styles/article-body/responsive.web.js]: Yea I agree, I think a follow up PR would be good.

File: packages/article/src/article-body/article-body-row.js (hunk 1 above)
Comment (jimhunty) [reply packages/article/src/article-body/article-body-row.js]: Sounds good, was tidying up what was already there. But can improve it further.

Comment (jimhunty) [reply packages/article/src/article-body/article-body-row.js]: Done.

**Pull request merged**